// groupdocs-viewer-cloud/lib/configuration.js
"use strict";
/*
* The MIT License (MIT)
*
* Copyright (c) 2003-2018 Aspose Pty Ltd
*
* Permission is hereby granted, free of charge, to any person obtaining a copy
* of this software and associated documentation files (the "Software"), to deal
* in the Software without restriction, including without limitation the rights
* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
* copies of the Software, and to permit persons to whom the Software is
* furnished to do so, subject to the following conditions:
*
* The above copyright notice and this permission notice shall be included in all
* copies or substantial portions of the Software.
*
* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
* SOFTWARE.
*/
Object.defineProperty(exports, "__esModule", { value: true });
const auth_1 = require("./auth");
/**
 * Words API configuration
 */
class Configuration {
    /**
     * @param appSid Application identifier (App SID)
     * @param appKey Application private key (App Key)
     */
    constructor(appSid, appKey) {
        /**
         * API base URL, default is 'https://api.groupdocs.cloud'
         */
        this.apiBaseUrl = "https://api.groupdocs.cloud";
        /**
         * Gets or sets the API version
         */
        this.apiVersion = "/v1";
        this.appSid = appSid;
        this.appKey = appKey;
        this.authentication = new auth_1.OAuth();
    }
    /**
     * Returns server url e.g "https://api.groupdocs.cloud/v1"
     */
    getServerUrl() {
        return this.apiBaseUrl + this.apiVersion;
    }
}
exports.Configuration = Configuration;
