// groupdocs-viewer-cloud/lib/viewer_api.js
"use strict";
/*
* The MIT License (MIT)
*
* Copyright (c) 2003-2018 Aspose Pty Ltd
*
* Permission is hereby granted, free of charge, to any person obtaining a copy
* of this software and associated documentation files (the "Software"), to deal
* in the Software without restriction, including without limitation the rights
* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
* copies of the Software, and to permit persons to whom the Software is
* furnished to do so, subject to the following conditions:
*
* The above copyright notice and this permission notice shall be included in all
* copies or substantial portions of the Software.
*
* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
* SOFTWARE.
*/
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
const api_client_1 = require("./api_client");
const configuration_1 = require("./configuration");
const serializer_1 = require("./serializer");
__export(require("./model"));
__export(require("./configuration"));
/**
 * GroupDocs.Viewer Cloud API
 */
class ViewerApi {
    /**
     * Creates new instance of ViewerApi
     * @param appSid Application identifier (App SID).
     * @param appKey Application private key (App Key).
     */
    static fromKeys(appSid, appKey) {
        const config = new configuration_1.Configuration(appSid, appKey);
        return new ViewerApi(config);
    }
    /**
     * Creates new instance of ViewerApi
     * @param config API configuration.
     */
    static fromConfig(config) {
        return new ViewerApi(config);
    }
    /**
     * @param config Configuration.
     */
    constructor(config) {
        this.configuration = config;
    }
    /**
     * Removes fonts cache.
     * @param requestObj contains request parameters
     */
    deleteFontsCache() {
        return __awaiter(this, void 0, void 0, function* () {
            const localVarPath = this.configuration.getServerUrl() + "/viewer/fonts/cache";
            const queryParameters = {};
            const requestOptions = {
                method: "DELETE",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            return Promise.resolve(response);
        });
    }
    /**
     * List installed fonts.
     * @param requestObj contains request parameters
     */
    getFonts() {
        return __awaiter(this, void 0, void 0, function* () {
            const localVarPath = this.configuration.getServerUrl() + "/viewer/fonts";
            const queryParameters = {};
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "FontCollection");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves list of supported file formats.
     * @param requestObj contains request parameters
     */
    getSupportedFileFormats() {
        return __awaiter(this, void 0, void 0, function* () {
            const localVarPath = this.configuration.getServerUrl() + "/viewer/formats";
            const queryParameters = {};
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "FormatCollection");
            return Promise.resolve(result);
        });
    }
    /**
     * Creates attachment cache.
     * @param requestObj contains request parameters
     */
    htmlCreateAttachmentPagesCache(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlCreateAttachmentPagesCache.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/attachments/{attachmentName}/pages"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "attachmentName" + "}", String(requestObj.attachmentName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlCreateAttachmentPagesCache.');
            }
            // verify required parameter 'requestObj.attachmentName' is not null or undefined
            if (requestObj.attachmentName === null || requestObj.attachmentName === undefined) {
                throw new Error('Required parameter "requestObj.attachmentName" was null or undefined when calling htmlCreateAttachmentPagesCache.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "POST",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
                body: serializer_1.Serializer.serialize(requestObj.htmlOptions, requestObj.htmlOptions.constructor.name === "Object" ? "HtmlOptions" : requestObj.htmlOptions.constructor.name),
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "HtmlAttachmentPageCollection");
            return Promise.resolve(result);
        });
    }
    /**
     * Creates document pages as HTML and saves them in cache. Pages created before will be removed from cache.
     * @param requestObj contains request parameters
     */
    htmlCreatePagesCache(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlCreatePagesCache.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/pages"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlCreatePagesCache.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "POST",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
                body: serializer_1.Serializer.serialize(requestObj.htmlOptions, requestObj.htmlOptions.constructor.name === "Object" ? "HtmlOptions" : requestObj.htmlOptions.constructor.name),
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "HtmlPageCollection");
            return Promise.resolve(result);
        });
    }
    /**
     * Creates posted document pages as HTML and saves them in cache. Content with document or multipart content is expected where first part is document and second is HtmlOptions. Saves retrieved file in storage. Use fileName and folder parameters to specify desired file name and folder to save file. When file with specified name already exists in storage new unique file name will be used for file.
     * @param requestObj contains request parameters
     */
    htmlCreatePagesCacheFromContent(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlCreatePagesCacheFromContent.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/html/pages";
            const queryParameters = {};
            const formParams = {};
            // verify required parameter 'requestObj.file' is not null or undefined
            if (requestObj.file === null || requestObj.file === undefined) {
                throw new Error('Required parameter "requestObj.file" was null or undefined when calling htmlCreatePagesCacheFromContent.');
            }
            // verify required parameter 'requestObj.htmlOptions' is not null or undefined
            if (requestObj.htmlOptions === null || requestObj.htmlOptions === undefined) {
                throw new Error('Required parameter "requestObj.htmlOptions" was null or undefined when calling htmlCreatePagesCacheFromContent.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fileName", requestObj.fileName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            if (requestObj.file !== undefined) {
                formParams.File = requestObj.file;
            }
            if (requestObj.htmlOptions !== undefined) {
                formParams.HtmlOptions = requestObj.htmlOptions;
            }
            const requestOptions = {
                method: "POST",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            requestOptions.formData = formParams;
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "HtmlPageCollection");
            return Promise.resolve(result);
        });
    }
    /**
     * Creates pages as HTML and saves them in cache for document at provided URL. Retrieves file from specified URL and tries to detect file type when fileName parameter is not specified. Saves retrieved file in storage. Use fileName and folder parameters to specify desired file name and folder to save file. When file with specified name already exists in storage new unique file name will be used for file.
     * @param requestObj contains request parameters
     */
    htmlCreatePagesCacheFromUrl(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlCreatePagesCacheFromUrl.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/html/pages";
            const queryParameters = {};
            // verify required parameter 'requestObj.url' is not null or undefined
            if (requestObj.url === null || requestObj.url === undefined) {
                throw new Error('Required parameter "requestObj.url" was null or undefined when calling htmlCreatePagesCacheFromUrl.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "url", requestObj.url);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fileName", requestObj.fileName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "PUT",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
                body: serializer_1.Serializer.serialize(requestObj.htmlOptions, requestObj.htmlOptions.constructor.name === "Object" ? "HtmlOptions" : requestObj.htmlOptions.constructor.name),
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "HtmlPageCollection");
            return Promise.resolve(result);
        });
    }
    /**
     * Creates PDF document.  Removes PDF document if it was created before.
     * @param requestObj contains request parameters
     */
    htmlCreatePdfFile(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlCreatePdfFile.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/pdf"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlCreatePdfFile.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "POST",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
                body: serializer_1.Serializer.serialize(requestObj.pdfFileOptions, requestObj.pdfFileOptions.constructor.name === "Object" ? "PdfFileOptions" : requestObj.pdfFileOptions.constructor.name),
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "PdfFileInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Creates PDF document from document passed in request body and saves it in cache. Content with document or multipart content is expected where first part is document and second is PdfFileOptions. Saves retrieved file in storage. Use fileName and folder parameters to specify desired file name and folder to save file. When file with specified name already exists in storage new unique file name will be used for file.
     * @param requestObj contains request parameters
     */
    htmlCreatePdfFileFromContent(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlCreatePdfFileFromContent.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/html/pdf";
            const queryParameters = {};
            const formParams = {};
            // verify required parameter 'requestObj.file' is not null or undefined
            if (requestObj.file === null || requestObj.file === undefined) {
                throw new Error('Required parameter "requestObj.file" was null or undefined when calling htmlCreatePdfFileFromContent.');
            }
            // verify required parameter 'requestObj.pdfFileOptions' is not null or undefined
            if (requestObj.pdfFileOptions === null || requestObj.pdfFileOptions === undefined) {
                throw new Error('Required parameter "requestObj.pdfFileOptions" was null or undefined when calling htmlCreatePdfFileFromContent.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fileName", requestObj.fileName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            if (requestObj.file !== undefined) {
                formParams.File = requestObj.file;
            }
            if (requestObj.pdfFileOptions !== undefined) {
                formParams.PdfFileOptions = requestObj.pdfFileOptions;
            }
            const requestOptions = {
                method: "POST",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            requestOptions.formData = formParams;
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "PdfFileInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Creates PDF document for document at provided URL and saves it in cache.  Retrieves file from specified URL and tries to detect file type when fileName parameter is not specified. Saves retrieved file in storage. Use fileName and folder parameters to specify desired file name and folder to save file. When file with specified name already exists in storage new unique file name will be used for file. Expects PdfFileOptions object data in request body.
     * @param requestObj contains request parameters
     */
    htmlCreatePdfFileFromUrl(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlCreatePdfFileFromUrl.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/html/pdf";
            const queryParameters = {};
            // verify required parameter 'requestObj.url' is not null or undefined
            if (requestObj.url === null || requestObj.url === undefined) {
                throw new Error('Required parameter "requestObj.url" was null or undefined when calling htmlCreatePdfFileFromUrl.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "url", requestObj.url);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fileName", requestObj.fileName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "PUT",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
                body: serializer_1.Serializer.serialize(requestObj.pdfFileOptions, requestObj.pdfFileOptions.constructor.name === "Object" ? "PdfFileOptions" : requestObj.pdfFileOptions.constructor.name),
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "PdfFileInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Removes attachment cache.
     * @param requestObj contains request parameters
     */
    htmlDeleteAttachmentPagesCache(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlDeleteAttachmentPagesCache.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/attachments/{attachmentName}/pages/cache"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "attachmentName" + "}", String(requestObj.attachmentName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlDeleteAttachmentPagesCache.');
            }
            // verify required parameter 'requestObj.attachmentName' is not null or undefined
            if (requestObj.attachmentName === null || requestObj.attachmentName === undefined) {
                throw new Error('Required parameter "requestObj.attachmentName" was null or undefined when calling htmlDeleteAttachmentPagesCache.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "DELETE",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            return Promise.resolve(response);
        });
    }
    /**
     * Removes document cache.
     * @param requestObj contains request parameters
     */
    htmlDeletePagesCache(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlDeletePagesCache.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/pages/cache"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlDeletePagesCache.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "DELETE",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            return Promise.resolve(response);
        });
    }
    /**
     * Downloads attachment.
     * @param requestObj contains request parameters
     */
    htmlGetAttachment(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetAttachment.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/attachments/{attachmentName}"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "attachmentName" + "}", String(requestObj.attachmentName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlGetAttachment.');
            }
            // verify required parameter 'requestObj.attachmentName' is not null or undefined
            if (requestObj.attachmentName === null || requestObj.attachmentName === undefined) {
                throw new Error('Required parameter "requestObj.attachmentName" was null or undefined when calling htmlGetAttachment.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                encoding: null,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "Buffer");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves attachment information.
     * @param requestObj contains request parameters
     */
    htmlGetAttachmentInfo(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetAttachmentInfo.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/attachments/{attachmentName}/info"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "attachmentName" + "}", String(requestObj.attachmentName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlGetAttachmentInfo.');
            }
            // verify required parameter 'requestObj.attachmentName' is not null or undefined
            if (requestObj.attachmentName === null || requestObj.attachmentName === undefined) {
                throw new Error('Required parameter "requestObj.attachmentName" was null or undefined when calling htmlGetAttachmentInfo.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "attachmentPassword", requestObj.attachmentPassword);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "DocumentInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves attachment information with specified DocumentInfoOptions. Expects DocumentInfoOptions object data in request body.
     * @param requestObj contains request parameters
     */
    htmlGetAttachmentInfoWithOptions(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetAttachmentInfoWithOptions.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/attachments/{attachmentName}/info"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "attachmentName" + "}", String(requestObj.attachmentName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlGetAttachmentInfoWithOptions.');
            }
            // verify required parameter 'requestObj.attachmentName' is not null or undefined
            if (requestObj.attachmentName === null || requestObj.attachmentName === undefined) {
                throw new Error('Required parameter "requestObj.attachmentName" was null or undefined when calling htmlGetAttachmentInfoWithOptions.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "POST",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
                body: serializer_1.Serializer.serialize(requestObj.documentInfoOptions, requestObj.documentInfoOptions.constructor.name === "Object" ? "DocumentInfoOptions" : requestObj.documentInfoOptions.constructor.name),
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "DocumentInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Downloads attachment page as HTML.
     * @param requestObj contains request parameters
     */
    htmlGetAttachmentPage(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetAttachmentPage.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/attachments/{attachmentName}/pages/{pageNumber}"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "attachmentName" + "}", String(requestObj.attachmentName))
                .replace("{" + "pageNumber" + "}", String(requestObj.pageNumber));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlGetAttachmentPage.');
            }
            // verify required parameter 'requestObj.attachmentName' is not null or undefined
            if (requestObj.attachmentName === null || requestObj.attachmentName === undefined) {
                throw new Error('Required parameter "requestObj.attachmentName" was null or undefined when calling htmlGetAttachmentPage.');
            }
            // verify required parameter 'requestObj.pageNumber' is not null or undefined
            if (requestObj.pageNumber === null || requestObj.pageNumber === undefined) {
                throw new Error('Required parameter "requestObj.pageNumber" was null or undefined when calling htmlGetAttachmentPage.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "resourcePath", requestObj.resourcePath);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "ignoreResourcePathInResources", requestObj.ignoreResourcePathInResources);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "embedResources", requestObj.embedResources);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "enableMinification", requestObj.enableMinification);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "enableResponsiveRendering", requestObj.enableResponsiveRendering);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "excludeFonts", requestObj.excludeFonts);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "attachmentPassword", requestObj.attachmentPassword);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "defaultFontName", requestObj.defaultFontName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                encoding: null,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "Buffer");
            return Promise.resolve(result);
        });
    }
    /**
     * Downloads HTML page resource (image, style, graphics or font).
     * @param requestObj contains request parameters
     */
    htmlGetAttachmentPageResource(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetAttachmentPageResource.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/attachments/{attachmentName}/pages/{pageNumber}/resources/{resourceName}"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "attachmentName" + "}", String(requestObj.attachmentName))
                .replace("{" + "pageNumber" + "}", String(requestObj.pageNumber))
                .replace("{" + "resourceName" + "}", String(requestObj.resourceName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlGetAttachmentPageResource.');
            }
            // verify required parameter 'requestObj.attachmentName' is not null or undefined
            if (requestObj.attachmentName === null || requestObj.attachmentName === undefined) {
                throw new Error('Required parameter "requestObj.attachmentName" was null or undefined when calling htmlGetAttachmentPageResource.');
            }
            // verify required parameter 'requestObj.pageNumber' is not null or undefined
            if (requestObj.pageNumber === null || requestObj.pageNumber === undefined) {
                throw new Error('Required parameter "requestObj.pageNumber" was null or undefined when calling htmlGetAttachmentPageResource.');
            }
            // verify required parameter 'requestObj.resourceName' is not null or undefined
            if (requestObj.resourceName === null || requestObj.resourceName === undefined) {
                throw new Error('Required parameter "requestObj.resourceName" was null or undefined when calling htmlGetAttachmentPageResource.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                encoding: null,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "Buffer");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves attachment pages as HTML.
     * @param requestObj contains request parameters
     */
    htmlGetAttachmentPages(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetAttachmentPages.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/attachments/{attachmentName}/pages"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "attachmentName" + "}", String(requestObj.attachmentName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlGetAttachmentPages.');
            }
            // verify required parameter 'requestObj.attachmentName' is not null or undefined
            if (requestObj.attachmentName === null || requestObj.attachmentName === undefined) {
                throw new Error('Required parameter "requestObj.attachmentName" was null or undefined when calling htmlGetAttachmentPages.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "resourcePath", requestObj.resourcePath);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "ignoreResourcePathInResources", requestObj.ignoreResourcePathInResources);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "embedResources", requestObj.embedResources);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "enableMinification", requestObj.enableMinification);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "enableResponsiveRendering", requestObj.enableResponsiveRendering);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "excludeFonts", requestObj.excludeFonts);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "startPageNumber", requestObj.startPageNumber);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "countPages", requestObj.countPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "attachmentPassword", requestObj.attachmentPassword);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "defaultFontName", requestObj.defaultFontName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "HtmlAttachmentPageCollection");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves list of document attachments.
     * @param requestObj contains request parameters
     */
    htmlGetAttachments(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetAttachments.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/attachments"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlGetAttachments.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "AttachmentCollection");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves document information.
     * @param requestObj contains request parameters
     */
    htmlGetDocumentInfo(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetDocumentInfo.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/info"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlGetDocumentInfo.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "DocumentInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves document information for posted document. Content with document or multipart content is expected where first part is document and second is DocumentInfoOptions. Saves file in storage. Use fileName and folder parameters to specify desired file name and folder to save file. When file with specified name already exists in storage new unique file name will be used for file.
     * @param requestObj contains request parameters
     */
    htmlGetDocumentInfoFromContent(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetDocumentInfoFromContent.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/html/info";
            const queryParameters = {};
            const formParams = {};
            // verify required parameter 'requestObj.file' is not null or undefined
            if (requestObj.file === null || requestObj.file === undefined) {
                throw new Error('Required parameter "requestObj.file" was null or undefined when calling htmlGetDocumentInfoFromContent.');
            }
            // verify required parameter 'requestObj.documentInfoOptions' is not null or undefined
            if (requestObj.documentInfoOptions === null || requestObj.documentInfoOptions === undefined) {
                throw new Error('Required parameter "requestObj.documentInfoOptions" was null or undefined when calling htmlGetDocumentInfoFromContent.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fileName", requestObj.fileName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            if (requestObj.file !== undefined) {
                formParams.File = requestObj.file;
            }
            if (requestObj.documentInfoOptions !== undefined) {
                formParams.DocumentInfoOptions = requestObj.documentInfoOptions;
            }
            const requestOptions = {
                method: "POST",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            requestOptions.formData = formParams;
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "DocumentInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves document information for document at provided URL. Retrieves file from specified URL and tries to detect file type when fileName parameter is not specified. Saves retrieved file in storage. Use fileName and folder parameters to specify desired file name and folder to save file. When file with specified name already exists in storage new unique file name will be used for file.
     * @param requestObj contains request parameters
     */
    htmlGetDocumentInfoFromUrl(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetDocumentInfoFromUrl.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/html/info";
            const queryParameters = {};
            // verify required parameter 'requestObj.url' is not null or undefined
            if (requestObj.url === null || requestObj.url === undefined) {
                throw new Error('Required parameter "requestObj.url" was null or undefined when calling htmlGetDocumentInfoFromUrl.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "url", requestObj.url);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fileName", requestObj.fileName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "DocumentInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves document information for document at provided URL. Retrieves file from specified URL and tries to detect file type when fileName parameter is not specified. Saves retrieved file in storage. Use fileName and folder parameters to specify desired file name and folder to save file. When file with specified name already exists in storage new unique file name will be used for file.
     * @param requestObj contains request parameters
     */
    htmlGetDocumentInfoFromUrlWithOptions(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetDocumentInfoFromUrlWithOptions.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/html/info";
            const queryParameters = {};
            // verify required parameter 'requestObj.url' is not null or undefined
            if (requestObj.url === null || requestObj.url === undefined) {
                throw new Error('Required parameter "requestObj.url" was null or undefined when calling htmlGetDocumentInfoFromUrlWithOptions.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "url", requestObj.url);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fileName", requestObj.fileName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "PUT",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
                body: serializer_1.Serializer.serialize(requestObj.documentInfoOptions, requestObj.documentInfoOptions.constructor.name === "Object" ? "DocumentInfoOptions" : requestObj.documentInfoOptions.constructor.name),
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "DocumentInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves document information with specified DocumentInfoOptions. Expects DocumentInfoOptions object data in request body.
     * @param requestObj contains request parameters
     */
    htmlGetDocumentInfoWithOptions(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetDocumentInfoWithOptions.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/info"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlGetDocumentInfoWithOptions.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "POST",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
                body: serializer_1.Serializer.serialize(requestObj.documentInfoOptions, requestObj.documentInfoOptions.constructor.name === "Object" ? "DocumentInfoOptions" : requestObj.documentInfoOptions.constructor.name),
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "DocumentInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Downloads document page as HTML.
     * @param requestObj contains request parameters
     */
    htmlGetPage(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetPage.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/pages/{pageNumber}"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "pageNumber" + "}", String(requestObj.pageNumber));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlGetPage.');
            }
            // verify required parameter 'requestObj.pageNumber' is not null or undefined
            if (requestObj.pageNumber === null || requestObj.pageNumber === undefined) {
                throw new Error('Required parameter "requestObj.pageNumber" was null or undefined when calling htmlGetPage.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "resourcePath", requestObj.resourcePath);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "ignoreResourcePathInResources", requestObj.ignoreResourcePathInResources);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "embedResources", requestObj.embedResources);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "enableMinification", requestObj.enableMinification);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "enableResponsiveRendering", requestObj.enableResponsiveRendering);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "excludeFonts", requestObj.excludeFonts);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "defaultFontName", requestObj.defaultFontName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                encoding: null,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "Buffer");
            return Promise.resolve(result);
        });
    }
    /**
     * Downloads HTML page resource (image, style, graphics or font).
     * @param requestObj contains request parameters
     */
    htmlGetPageResource(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetPageResource.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/pages/{pageNumber}/resources/{resourceName}"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "pageNumber" + "}", String(requestObj.pageNumber))
                .replace("{" + "resourceName" + "}", String(requestObj.resourceName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlGetPageResource.');
            }
            // verify required parameter 'requestObj.pageNumber' is not null or undefined
            if (requestObj.pageNumber === null || requestObj.pageNumber === undefined) {
                throw new Error('Required parameter "requestObj.pageNumber" was null or undefined when calling htmlGetPageResource.');
            }
            // verify required parameter 'requestObj.resourceName' is not null or undefined
            if (requestObj.resourceName === null || requestObj.resourceName === undefined) {
                throw new Error('Required parameter "requestObj.resourceName" was null or undefined when calling htmlGetPageResource.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                encoding: null,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "Buffer");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves list of document pages as HTML.
     * @param requestObj contains request parameters
     */
    htmlGetPages(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetPages.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/pages"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlGetPages.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "resourcePath", requestObj.resourcePath);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "ignoreResourcePathInResources", requestObj.ignoreResourcePathInResources);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "embedResources", requestObj.embedResources);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "enableMinification", requestObj.enableMinification);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "enableResponsiveRendering", requestObj.enableResponsiveRendering);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "excludeFonts", requestObj.excludeFonts);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "startPageNumber", requestObj.startPageNumber);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "countPages", requestObj.countPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "defaultFontName", requestObj.defaultFontName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "HtmlPageCollection");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves list of document pages as HTML.
     * @param requestObj contains request parameters
     */
    htmlGetPagesFromUrl(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetPagesFromUrl.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/html/pages";
            const queryParameters = {};
            // verify required parameter 'requestObj.url' is not null or undefined
            if (requestObj.url === null || requestObj.url === undefined) {
                throw new Error('Required parameter "requestObj.url" was null or undefined when calling htmlGetPagesFromUrl.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "url", requestObj.url);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fileName", requestObj.fileName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "resourcePath", requestObj.resourcePath);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "ignoreResourcePathInResources", requestObj.ignoreResourcePathInResources);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "embedResources", requestObj.embedResources);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "enableMinification", requestObj.enableMinification);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "enableResponsiveRendering", requestObj.enableResponsiveRendering);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "excludeFonts", requestObj.excludeFonts);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "startPageNumber", requestObj.startPageNumber);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "countPages", requestObj.countPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "defaultFontName", requestObj.defaultFontName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "HtmlPageCollection");
            return Promise.resolve(result);
        });
    }
    /**
     * Downloads document as PDF.
     * @param requestObj contains request parameters
     */
    htmlGetPdfFile(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetPdfFile.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/pdf"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlGetPdfFile.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "defaultFontName", requestObj.defaultFontName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                encoding: null,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "Buffer");
            return Promise.resolve(result);
        });
    }
    /**
     * Downloads document at provided URL as PDF.  Document will be retrieved from the passed URL and added to Storage.
     * @param requestObj contains request parameters
     */
    htmlGetPdfFileFromUrl(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetPdfFileFromUrl.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/html/pdf";
            const queryParameters = {};
            // verify required parameter 'requestObj.url' is not null or undefined
            if (requestObj.url === null || requestObj.url === undefined) {
                throw new Error('Required parameter "requestObj.url" was null or undefined when calling htmlGetPdfFileFromUrl.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "url", requestObj.url);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fileName", requestObj.fileName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "defaultFontName", requestObj.defaultFontName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                encoding: null,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "Buffer");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves attachment pages as HTML.
     * @param requestObj contains request parameters
     */
    htmlGetZipWithAttachmentPages(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetZipWithAttachmentPages.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/attachments/{attachmentName}/pages/zip"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "attachmentName" + "}", String(requestObj.attachmentName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlGetZipWithAttachmentPages.');
            }
            // verify required parameter 'requestObj.attachmentName' is not null or undefined
            if (requestObj.attachmentName === null || requestObj.attachmentName === undefined) {
                throw new Error('Required parameter "requestObj.attachmentName" was null or undefined when calling htmlGetZipWithAttachmentPages.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "resourcePath", requestObj.resourcePath);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "ignoreResourcePathInResources", requestObj.ignoreResourcePathInResources);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "embedResources", requestObj.embedResources);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "enableMinification", requestObj.enableMinification);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "enableResponsiveRendering", requestObj.enableResponsiveRendering);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "excludeFonts", requestObj.excludeFonts);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "startPageNumber", requestObj.startPageNumber);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "countPages", requestObj.countPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "attachmentPassword", requestObj.attachmentPassword);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "defaultFontName", requestObj.defaultFontName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                encoding: null,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "Buffer");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves list of document pages as HTML.
     * @param requestObj contains request parameters
     */
    htmlGetZipWithPages(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlGetZipWithPages.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/pages/zip"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlGetZipWithPages.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "resourcePath", requestObj.resourcePath);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "ignoreResourcePathInResources", requestObj.ignoreResourcePathInResources);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "embedResources", requestObj.embedResources);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "enableMinification", requestObj.enableMinification);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "enableResponsiveRendering", requestObj.enableResponsiveRendering);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "excludeFonts", requestObj.excludeFonts);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "startPageNumber", requestObj.startPageNumber);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "countPages", requestObj.countPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "defaultFontName", requestObj.defaultFontName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                encoding: null,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "Buffer");
            return Promise.resolve(result);
        });
    }
    /**
     * Rotates or reorders document page(s).
     * @param requestObj contains request parameters
     */
    htmlTransformPages(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling htmlTransformPages.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/html/pages"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling htmlTransformPages.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "PUT",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
                body: serializer_1.Serializer.serialize(requestObj.transformOptions, requestObj.transformOptions.constructor.name === "Object" ? "TransformOptionsBase" : requestObj.transformOptions.constructor.name),
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "PageInfoCollection");
            return Promise.resolve(result);
        });
    }
    /**
     * Creates attachment cache.
     * @param requestObj contains request parameters
     */
    imageCreateAttachmentPagesCache(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageCreateAttachmentPagesCache.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/image/attachments/{attachmentName}/pages"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "attachmentName" + "}", String(requestObj.attachmentName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling imageCreateAttachmentPagesCache.');
            }
            // verify required parameter 'requestObj.attachmentName' is not null or undefined
            if (requestObj.attachmentName === null || requestObj.attachmentName === undefined) {
                throw new Error('Required parameter "requestObj.attachmentName" was null or undefined when calling imageCreateAttachmentPagesCache.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "POST",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
                body: serializer_1.Serializer.serialize(requestObj.imageOptions, requestObj.imageOptions.constructor.name === "Object" ? "ImageOptions" : requestObj.imageOptions.constructor.name),
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "ImageAttachmentPageCollection");
            return Promise.resolve(result);
        });
    }
    /**
     * Creates document pages as image and saves them in cache.  Pages created before will be removed from cache.
     * @param requestObj contains request parameters
     */
    imageCreatePagesCache(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageCreatePagesCache.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/image/pages"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling imageCreatePagesCache.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "POST",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
                body: serializer_1.Serializer.serialize(requestObj.imageOptions, requestObj.imageOptions.constructor.name === "Object" ? "ImageOptions" : requestObj.imageOptions.constructor.name),
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "ImagePageCollection");
            return Promise.resolve(result);
        });
    }
    /**
     * Creates posted document pages as image and saves them in cache. Content with document or multipart content is expected where first part is document and second is HtmlOptions. Saves retrieved file in storage. Use fileName and folder parameters to specify desired file name and folder to save file. When file with specified name already exists in storage new unique file name will be used for file.
     * @param requestObj contains request parameters
     */
    imageCreatePagesCacheFromContent(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageCreatePagesCacheFromContent.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/image/pages";
            const queryParameters = {};
            const formParams = {};
            // verify required parameter 'requestObj.file' is not null or undefined
            if (requestObj.file === null || requestObj.file === undefined) {
                throw new Error('Required parameter "requestObj.file" was null or undefined when calling imageCreatePagesCacheFromContent.');
            }
            // verify required parameter 'requestObj.imageOptions' is not null or undefined
            if (requestObj.imageOptions === null || requestObj.imageOptions === undefined) {
                throw new Error('Required parameter "requestObj.imageOptions" was null or undefined when calling imageCreatePagesCacheFromContent.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fileName", requestObj.fileName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            if (requestObj.file !== undefined) {
                formParams.File = requestObj.file;
            }
            if (requestObj.imageOptions !== undefined) {
                formParams.ImageOptions = requestObj.imageOptions;
            }
            const requestOptions = {
                method: "POST",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            requestOptions.formData = formParams;
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "ImagePageCollection");
            return Promise.resolve(result);
        });
    }
    /**
     * Creates pages as image and saves them in cache for document at provided URL. Retrieves file from specified URL and tries to detect file type when fileName parameter is not specified. Saves retrieved file in storage. Use fileName and folder parameters to specify desired file name and folder to save file. When file with specified name already exists in storage new unique file name will be used for file.
     * @param requestObj contains request parameters
     */
    imageCreatePagesCacheFromUrl(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageCreatePagesCacheFromUrl.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/image/pages";
            const queryParameters = {};
            // verify required parameter 'requestObj.url' is not null or undefined
            if (requestObj.url === null || requestObj.url === undefined) {
                throw new Error('Required parameter "requestObj.url" was null or undefined when calling imageCreatePagesCacheFromUrl.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "url", requestObj.url);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fileName", requestObj.fileName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "PUT",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
                body: serializer_1.Serializer.serialize(requestObj.imageOptions, requestObj.imageOptions.constructor.name === "Object" ? "ImageOptions" : requestObj.imageOptions.constructor.name),
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "ImagePageCollection");
            return Promise.resolve(result);
        });
    }
    /**
     * Creates PDF document.  Removes PDF document if it was created before.
     * @param requestObj contains request parameters
     */
    imageCreatePdfFile(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageCreatePdfFile.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/image/pdf"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling imageCreatePdfFile.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "POST",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
                body: serializer_1.Serializer.serialize(requestObj.pdfFileOptions, requestObj.pdfFileOptions.constructor.name === "Object" ? "PdfFileOptions" : requestObj.pdfFileOptions.constructor.name),
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "PdfFileInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Creates PDF document from document passed in request body and saves it in cache. Content with document or multipart content is expected where first part is document and second is PdfFileOptions. Saves retrieved file in storage. Use fileName and folder parameters to specify desired file name and folder to save file. When file with specified name already exists in storage new unique file name will be used for file.
     * @param requestObj contains request parameters
     */
    imageCreatePdfFileFromContent(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageCreatePdfFileFromContent.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/image/pdf";
            const queryParameters = {};
            const formParams = {};
            // verify required parameter 'requestObj.file' is not null or undefined
            if (requestObj.file === null || requestObj.file === undefined) {
                throw new Error('Required parameter "requestObj.file" was null or undefined when calling imageCreatePdfFileFromContent.');
            }
            // verify required parameter 'requestObj.pdfFileOptions' is not null or undefined
            if (requestObj.pdfFileOptions === null || requestObj.pdfFileOptions === undefined) {
                throw new Error('Required parameter "requestObj.pdfFileOptions" was null or undefined when calling imageCreatePdfFileFromContent.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fileName", requestObj.fileName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            if (requestObj.file !== undefined) {
                formParams.File = requestObj.file;
            }
            if (requestObj.pdfFileOptions !== undefined) {
                formParams.PdfFileOptions = requestObj.pdfFileOptions;
            }
            const requestOptions = {
                method: "POST",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            requestOptions.formData = formParams;
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "PdfFileInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Creates PDF document for document at provided URL and saves it in cache.  Retrieves file from specified URL and tries to detect file type when fileName parameter is not specified. Saves retrieved file in storage. Use fileName and folder parameters to specify desired file name and folder to save file. When file with specified name already exists in storage new unique file name will be used for file. Expects PdfFileOptions object data in request body.
     * @param requestObj contains request parameters
     */
    imageCreatePdfFileFromUrl(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageCreatePdfFileFromUrl.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/image/pdf";
            const queryParameters = {};
            // verify required parameter 'requestObj.url' is not null or undefined
            if (requestObj.url === null || requestObj.url === undefined) {
                throw new Error('Required parameter "requestObj.url" was null or undefined when calling imageCreatePdfFileFromUrl.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "url", requestObj.url);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fileName", requestObj.fileName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "PUT",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
                body: serializer_1.Serializer.serialize(requestObj.pdfFileOptions, requestObj.pdfFileOptions.constructor.name === "Object" ? "PdfFileOptions" : requestObj.pdfFileOptions.constructor.name),
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "PdfFileInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Removes attachment cache.
     * @param requestObj contains request parameters
     */
    imageDeleteAttachmentPagesCache(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageDeleteAttachmentPagesCache.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/image/attachments/{attachmentName}/pages/cache"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "attachmentName" + "}", String(requestObj.attachmentName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling imageDeleteAttachmentPagesCache.');
            }
            // verify required parameter 'requestObj.attachmentName' is not null or undefined
            if (requestObj.attachmentName === null || requestObj.attachmentName === undefined) {
                throw new Error('Required parameter "requestObj.attachmentName" was null or undefined when calling imageDeleteAttachmentPagesCache.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "DELETE",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            return Promise.resolve(response);
        });
    }
    /**
     * Removes document cache.
     * @param requestObj contains request parameters
     */
    imageDeletePagesCache(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageDeletePagesCache.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/image/pages/cache"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling imageDeletePagesCache.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "DELETE",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            return Promise.resolve(response);
        });
    }
    /**
     * Downloads attachment.
     * @param requestObj contains request parameters
     */
    imageGetAttachment(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageGetAttachment.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/image/attachments/{attachmentName}"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "attachmentName" + "}", String(requestObj.attachmentName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling imageGetAttachment.');
            }
            // verify required parameter 'requestObj.attachmentName' is not null or undefined
            if (requestObj.attachmentName === null || requestObj.attachmentName === undefined) {
                throw new Error('Required parameter "requestObj.attachmentName" was null or undefined when calling imageGetAttachment.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                encoding: null,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "Buffer");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves attachment information.
     * @param requestObj contains request parameters
     */
    imageGetAttachmentInfo(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageGetAttachmentInfo.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/image/attachments/{attachmentName}/info"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "attachmentName" + "}", String(requestObj.attachmentName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling imageGetAttachmentInfo.');
            }
            // verify required parameter 'requestObj.attachmentName' is not null or undefined
            if (requestObj.attachmentName === null || requestObj.attachmentName === undefined) {
                throw new Error('Required parameter "requestObj.attachmentName" was null or undefined when calling imageGetAttachmentInfo.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "extractText", requestObj.extractText);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "attachmentPassword", requestObj.attachmentPassword);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "DocumentInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves attachment information with specified DocumentInfoOptions. Expects DocumentInfoOptions object data in request body.
     * @param requestObj contains request parameters
     */
    imageGetAttachmentInfoWithOptions(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageGetAttachmentInfoWithOptions.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/image/attachments/{attachmentName}/info"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "attachmentName" + "}", String(requestObj.attachmentName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling imageGetAttachmentInfoWithOptions.');
            }
            // verify required parameter 'requestObj.attachmentName' is not null or undefined
            if (requestObj.attachmentName === null || requestObj.attachmentName === undefined) {
                throw new Error('Required parameter "requestObj.attachmentName" was null or undefined when calling imageGetAttachmentInfoWithOptions.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "POST",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
                body: serializer_1.Serializer.serialize(requestObj.documentInfoOptions, requestObj.documentInfoOptions.constructor.name === "Object" ? "DocumentInfoOptions" : requestObj.documentInfoOptions.constructor.name),
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "DocumentInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Downloads attachment page as image.
     * @param requestObj contains request parameters
     */
    imageGetAttachmentPage(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageGetAttachmentPage.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/image/attachments/{attachmentName}/pages/{pageNumber}"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "attachmentName" + "}", String(requestObj.attachmentName))
                .replace("{" + "pageNumber" + "}", String(requestObj.pageNumber));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling imageGetAttachmentPage.');
            }
            // verify required parameter 'requestObj.attachmentName' is not null or undefined
            if (requestObj.attachmentName === null || requestObj.attachmentName === undefined) {
                throw new Error('Required parameter "requestObj.attachmentName" was null or undefined when calling imageGetAttachmentPage.');
            }
            // verify required parameter 'requestObj.pageNumber' is not null or undefined
            if (requestObj.pageNumber === null || requestObj.pageNumber === undefined) {
                throw new Error('Required parameter "requestObj.pageNumber" was null or undefined when calling imageGetAttachmentPage.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "format", requestObj.format);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "width", requestObj.width);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "height", requestObj.height);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "quality", requestObj.quality);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "attachmentPassword", requestObj.attachmentPassword);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "extractText", requestObj.extractText);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "defaultFontName", requestObj.defaultFontName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                encoding: null,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "Buffer");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves attachment pages as images.
     * @param requestObj contains request parameters
     */
    imageGetAttachmentPages(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageGetAttachmentPages.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/image/attachments/{attachmentName}/pages"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "attachmentName" + "}", String(requestObj.attachmentName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling imageGetAttachmentPages.');
            }
            // verify required parameter 'requestObj.attachmentName' is not null or undefined
            if (requestObj.attachmentName === null || requestObj.attachmentName === undefined) {
                throw new Error('Required parameter "requestObj.attachmentName" was null or undefined when calling imageGetAttachmentPages.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "format", requestObj.format);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "width", requestObj.width);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "height", requestObj.height);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "quality", requestObj.quality);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "startPageNumber", requestObj.startPageNumber);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "countPages", requestObj.countPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "attachmentPassword", requestObj.attachmentPassword);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "extractText", requestObj.extractText);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "defaultFontName", requestObj.defaultFontName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "ImageAttachmentPageCollection");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves list of document attachments.
     * @param requestObj contains request parameters
     */
    imageGetAttachments(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageGetAttachments.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/image/attachments"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling imageGetAttachments.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "AttachmentCollection");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves document information.
     * @param requestObj contains request parameters
     */
    imageGetDocumentInfo(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageGetDocumentInfo.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/image/info"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling imageGetDocumentInfo.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "extractText", requestObj.extractText);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "DocumentInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves document information for posted document. Content with document or multipart content is expected where first part is document and second is DocumentInfoOptions. Saves file in storage. Use fileName and folder parameters to specify desired file name and folder to save file. When file with specified name already exists in storage new unique file name will be used for file.
     * @param requestObj contains request parameters
     */
    imageGetDocumentInfoFromContent(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageGetDocumentInfoFromContent.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/image/info";
            const queryParameters = {};
            const formParams = {};
            // verify required parameter 'requestObj.file' is not null or undefined
            if (requestObj.file === null || requestObj.file === undefined) {
                throw new Error('Required parameter "requestObj.file" was null or undefined when calling imageGetDocumentInfoFromContent.');
            }
            // verify required parameter 'requestObj.documentInfoOptions' is not null or undefined
            if (requestObj.documentInfoOptions === null || requestObj.documentInfoOptions === undefined) {
                throw new Error('Required parameter "requestObj.documentInfoOptions" was null or undefined when calling imageGetDocumentInfoFromContent.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fileName", requestObj.fileName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            if (requestObj.file !== undefined) {
                formParams.File = requestObj.file;
            }
            if (requestObj.documentInfoOptions !== undefined) {
                formParams.DocumentInfoOptions = requestObj.documentInfoOptions;
            }
            const requestOptions = {
                method: "POST",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            requestOptions.formData = formParams;
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "DocumentInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves document information for document at provided URL. Retrieves file from specified URL and tries to detect file type when fileName parameter is not specified. Saves retrieved file in storage. Use fileName and folder parameters to specify desired file name and folder to save file. When file with specified name already exists in storage new unique file name will be used for file.
     * @param requestObj contains request parameters
     */
    imageGetDocumentInfoFromUrl(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageGetDocumentInfoFromUrl.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/image/info";
            const queryParameters = {};
            // verify required parameter 'requestObj.url' is not null or undefined
            if (requestObj.url === null || requestObj.url === undefined) {
                throw new Error('Required parameter "requestObj.url" was null or undefined when calling imageGetDocumentInfoFromUrl.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "url", requestObj.url);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fileName", requestObj.fileName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "extractText", requestObj.extractText);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "DocumentInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves document information for document at provided URL. Retrieves file from specified URL and tries to detect file type when fileName parameter is not specified. Saves retrieved file in storage. Use fileName and folder parameters to specify desired file name and folder to save file. When file with specified name already exists in storage new unique file name will be used for file.
     * @param requestObj contains request parameters
     */
    imageGetDocumentInfoFromUrlWithOptions(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageGetDocumentInfoFromUrlWithOptions.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/image/info";
            const queryParameters = {};
            // verify required parameter 'requestObj.url' is not null or undefined
            if (requestObj.url === null || requestObj.url === undefined) {
                throw new Error('Required parameter "requestObj.url" was null or undefined when calling imageGetDocumentInfoFromUrlWithOptions.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "url", requestObj.url);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fileName", requestObj.fileName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "PUT",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
                body: serializer_1.Serializer.serialize(requestObj.documentInfoOptions, requestObj.documentInfoOptions.constructor.name === "Object" ? "DocumentInfoOptions" : requestObj.documentInfoOptions.constructor.name),
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "DocumentInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves document information with specified options. Expects DocumentInfoOptions object data in request body.
     * @param requestObj contains request parameters
     */
    imageGetDocumentInfoWithOptions(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageGetDocumentInfoWithOptions.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/image/info"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling imageGetDocumentInfoWithOptions.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "POST",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
                body: serializer_1.Serializer.serialize(requestObj.documentInfoOptions, requestObj.documentInfoOptions.constructor.name === "Object" ? "DocumentInfoOptions" : requestObj.documentInfoOptions.constructor.name),
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "DocumentInfo");
            return Promise.resolve(result);
        });
    }
    /**
     * Downloads document page as image.
     * @param requestObj contains request parameters
     */
    imageGetPage(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageGetPage.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/image/pages/{pageNumber}"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "pageNumber" + "}", String(requestObj.pageNumber));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling imageGetPage.');
            }
            // verify required parameter 'requestObj.pageNumber' is not null or undefined
            if (requestObj.pageNumber === null || requestObj.pageNumber === undefined) {
                throw new Error('Required parameter "requestObj.pageNumber" was null or undefined when calling imageGetPage.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "format", requestObj.format);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "width", requestObj.width);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "height", requestObj.height);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "quality", requestObj.quality);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "extractText", requestObj.extractText);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "defaultFontName", requestObj.defaultFontName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                encoding: null,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "Buffer");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves list of document pages as image.
     * @param requestObj contains request parameters
     */
    imageGetPages(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageGetPages.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/image/pages"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling imageGetPages.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "format", requestObj.format);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "width", requestObj.width);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "height", requestObj.height);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "quality", requestObj.quality);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "startPageNumber", requestObj.startPageNumber);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "countPages", requestObj.countPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "extractText", requestObj.extractText);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "defaultFontName", requestObj.defaultFontName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "ImagePageCollection");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves list of document pages as image.
     * @param requestObj contains request parameters
     */
    imageGetPagesFromUrl(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageGetPagesFromUrl.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/image/pages";
            const queryParameters = {};
            // verify required parameter 'requestObj.url' is not null or undefined
            if (requestObj.url === null || requestObj.url === undefined) {
                throw new Error('Required parameter "requestObj.url" was null or undefined when calling imageGetPagesFromUrl.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "url", requestObj.url);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fileName", requestObj.fileName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "format", requestObj.format);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "width", requestObj.width);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "height", requestObj.height);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "quality", requestObj.quality);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "startPageNumber", requestObj.startPageNumber);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "countPages", requestObj.countPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "extractText", requestObj.extractText);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "defaultFontName", requestObj.defaultFontName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "ImagePageCollection");
            return Promise.resolve(result);
        });
    }
    /**
     * Downloads document as PDF.
     * @param requestObj contains request parameters
     */
    imageGetPdfFile(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageGetPdfFile.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/image/pdf"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling imageGetPdfFile.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "defaultFontName", requestObj.defaultFontName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                encoding: null,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "Buffer");
            return Promise.resolve(result);
        });
    }
    /**
     * Downloads document at provided URL as PDF.  Document will be retrieved from the passed URL and added to Storage.
     * @param requestObj contains request parameters
     */
    imageGetPdfFileFromUrl(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageGetPdfFileFromUrl.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/image/pdf";
            const queryParameters = {};
            // verify required parameter 'requestObj.url' is not null or undefined
            if (requestObj.url === null || requestObj.url === undefined) {
                throw new Error('Required parameter "requestObj.url" was null or undefined when calling imageGetPdfFileFromUrl.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "url", requestObj.url);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fileName", requestObj.fileName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "defaultFontName", requestObj.defaultFontName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                encoding: null,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "Buffer");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves attachment pages as images.
     * @param requestObj contains request parameters
     */
    imageGetZipWithAttachmentPages(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageGetZipWithAttachmentPages.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/image/attachments/{attachmentName}/pages/zip"
                .replace("{" + "fileName" + "}", String(requestObj.fileName))
                .replace("{" + "attachmentName" + "}", String(requestObj.attachmentName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling imageGetZipWithAttachmentPages.');
            }
            // verify required parameter 'requestObj.attachmentName' is not null or undefined
            if (requestObj.attachmentName === null || requestObj.attachmentName === undefined) {
                throw new Error('Required parameter "requestObj.attachmentName" was null or undefined when calling imageGetZipWithAttachmentPages.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "format", requestObj.format);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "width", requestObj.width);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "height", requestObj.height);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "quality", requestObj.quality);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "startPageNumber", requestObj.startPageNumber);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "countPages", requestObj.countPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "attachmentPassword", requestObj.attachmentPassword);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "extractText", requestObj.extractText);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "defaultFontName", requestObj.defaultFontName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                encoding: null,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "Buffer");
            return Promise.resolve(result);
        });
    }
    /**
     * Retrieves list of document pages as image.
     * @param requestObj contains request parameters
     */
    imageGetZipWithPages(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageGetZipWithPages.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/image/pages/zip"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling imageGetZipWithPages.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "format", requestObj.format);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "width", requestObj.width);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "height", requestObj.height);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "quality", requestObj.quality);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "startPageNumber", requestObj.startPageNumber);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "countPages", requestObj.countPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "password", requestObj.password);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "extractText", requestObj.extractText);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderComments", requestObj.renderComments);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "renderHiddenPages", requestObj.renderHiddenPages);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "defaultFontName", requestObj.defaultFontName);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "fontsFolder", requestObj.fontsFolder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "GET",
                qs: queryParameters,
                uri: localVarPath,
                encoding: null,
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "Buffer");
            return Promise.resolve(result);
        });
    }
    /**
     * Rotates or reorders document page(s).
     * @param requestObj contains request parameters
     */
    imageTransformPages(requestObj) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestObj === null || requestObj === undefined) {
                throw new Error('Required parameter "requestObj" was null or undefined when calling imageTransformPages.');
            }
            let localVarPath = this.configuration.getServerUrl() + "/viewer/{fileName}/image/pages"
                .replace("{" + "fileName" + "}", String(requestObj.fileName));
            const queryParameters = {};
            // verify required parameter 'requestObj.fileName' is not null or undefined
            if (requestObj.fileName === null || requestObj.fileName === undefined) {
                throw new Error('Required parameter "requestObj.fileName" was null or undefined when calling imageTransformPages.');
            }
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "folder", requestObj.folder);
            localVarPath = api_client_1.addQueryParameterToUrl(localVarPath, queryParameters, "storage", requestObj.storage);
            const requestOptions = {
                method: "PUT",
                qs: queryParameters,
                uri: localVarPath,
                json: true,
                body: serializer_1.Serializer.serialize(requestObj.transformOptions, requestObj.transformOptions.constructor.name === "Object" ? "TransformOptionsBase" : requestObj.transformOptions.constructor.name),
            };
            const response = yield api_client_1.invokeApiMethod(requestOptions, this.configuration);
            const result = serializer_1.Serializer.deserialize(response.body, "PageInfoCollection");
            return Promise.resolve(result);
        });
    }
}
exports.ViewerApi = ViewerApi;
